src/data/equiv/set.lean

```diff
@@ -338,7 +338,7 @@ protected def compl {α : Type u} {β : Type v} {s : set α} {t : set β} [decid
 
 /-- The set product of two sets is equivalent to the type product of their coercions to types. -/
 protected def prod {α β} (s : set α) (t : set β) :
-  s.prod t ≃ s × t :=
+  (s ×ˢ t : set _) ≃ s × t :=
```

Comment:
@eric-wieser Are you asking about this line? Unfortunately, I have to add `: set _`. Possibly, this can go away once we use Lean 3.37c with https://github.com/leanprover-community/lean/pull/659

Reply:
Can you remove `set _` if you add the coe_sort arrow instead?

Reply:
AFAIK, no but I'd prefer to postpone this discussion till we have 3.37

Reply:
From the title alone, I don't see how lean#659 can be relevant given that's about simp and this is about the elaborator. Perhaps I need to read it in more detail.

Reply:
Sorry, I meant https://github.com/leanprover-community/lean/pull/657 (not sure that it actually fixes this issue but it seems relevant).

Reply:
I tested, this works fine:
```suggestion
  ↥(s ×ˢ t) ≃ s × t :=
```